solidity/contracts/l2/L2TBTC.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// TODO:
+// * Be paused by any one of `n` guardians, allowing avoidance of contagion in case
+//  of a chain- or bridge-specific incident.
+// * Mint by minter.
+// * Proper documentation.
+// * Test initialization
+contract L2TBTC is ERC20Upgradeable, OwnableUpgradeable {
+    /// @notice Indicates if the given address is a Minter. Only Minters can
+    ///         mint the token.
+    mapping(address => bool) public isMinter;
+
+    /// @notice List of all Minters.
+    address[] public minters;
+
+    event MinterAdded(address indexed minter);
+    event MinterRemoved(address indexed minter);
+
+    modifier onlyMinter() {
+        require(isMinter[msg.sender], "Caller is not a minter");
+        _;
+    }
+
+    function initialize(string memory _name, string memory _symbol)
+        external
```

Comment:
My understanding was that this `L2TBTC` token is going to be a generic contract that actual tokens will inherit from for a given L2/sidechains like Arbitrum, Polygon etc. Shouldn't we change the visibility of `initialize` to `internal`?

Reply:
I think this approach is more flexible. We can inherit a contract and add no code. We can also not inherit a contract at all and just deploy `L2TBTC` instance. Contract deployment and initialization happen in one transaction (see `helpers.upgrades.deployProxy`) so there is no chance someone can initialize your contract with the wrong values even if they wanted to which is pretty unlikely.

Reply:
ok, I thought that we wanted the names hardcoded in the contract. I'm fine with the `initialize` approach.